ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.view.js

```diff
@@ -107,102 +124,41 @@ module.exports = Marionette.LayoutView.extend({
-            pointText = pointText.substring(0, pointText.length - 1)
-            var latLon = pointText.split(' ')
-            this.model.set({
-              mode: 'circle',
-              locationType: 'latlon',
-              lat: latLon[1],
-              lon: latLon[0],
-              radius: filter.distance,
-            })
-            wreqr.vent.trigger('search:circledisplay', this.model)
-          } else if (CQLUtils.isPolygonFilter(filter)) {
-            this.handlePolygonDeserialization(filter)
-          } else {
-            let pointText = filterValue.substring(11)
-            pointText = pointText.substring(0, pointText.length - 1)
-            this.model.set({
-              mode: 'line',
-              lineWidth: filter.distance,
-              line: pointText.split(',').map(function(coordinate) {
-                return coordinate.split(' ').map(function(value) {
-                  return Number(value)
-                })
-              }),
-            })
-            wreqr.vent.trigger('search:linedisplay', this.model)
-          }
-          break
-        case 'INTERSECTS':
-          if (!filterValue || typeof filterValue !== 'string') {
-            break
-          }
-          this.handlePolygonDeserialization({
-            polygon: CQLUtils.arrayFromPolygonWkt(filterValue),
-          })
-          break
-        // these cases are for when the model matches the location model
-        case 'BBOX':
-          this.model.set({
-            mode: 'bbox',
-            locationType: 'latlon',
-            north: filter.north,
-            south: filter.south,
-            east: filter.east,
-            west: filter.west,
-          })
-          wreqr.vent.trigger('search:bboxdisplay', this.model)
-          break
-        case 'MULTIPOLYGON':
-        case 'POLYGON':
-          this.handlePolygonDeserialization(filter)
-          break
-        case 'POINTRADIUS':
-          this.model.set({
-            mode: 'circle',
-            locationType: 'latlon',
-            lat: filter.lat,
-            lon: filter.lon,
-            radius: filter.radius,
-          })
+    if (!this.propertyModel) return
+
+    const filter = this.propertyModel.get('value')
+    this.model.set(filterToLocationOldModel(filter))
+
+    switch (filter.type) {
+      // these cases are for when the model matches the filter model
+      case 'DWITHIN':
+        if (CQLUtils.isPointRadiusFilter(filter)) {
           wreqr.vent.trigger('search:circledisplay', this.model)
-          break
-        case 'LINE':
-          this.model.set({
-            mode: 'line',
-            line: filter.line,
-            lineWidth: filter.lineWidth,
-          })
+        } else if (CQLUtils.isPolygonFilter(filter)) {
+          wreqr.vent.trigger('search:polydisplay', this.model)
+        } else {
           wreqr.vent.trigger('search:linedisplay', this.model)
-          break
-      }
+        }
+        break
+      case 'INTERSECTS':
```

Comment:
❓ Since all these do know is call `trigger` with a different string, perhaps you could create a map of `filter.type` to the `search:xxxxdisplay` string that they should use to call trigger. `DWITHIN` ruins that pattern a bit but it still might be worth it.